compiler/rustc_hir/src/hir_id.rs

```diff
@@ -44,18 +40,6 @@ impl fmt::Display for HirId {
     }
 }
 
-impl Ord for HirId {
-    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
-        (self.index()).cmp(&(other.index()))
-    }
-}
-
-impl PartialOrd for HirId {
-    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
-        Some(self.cmp(&other))
-    }
-}
-
```

Comment:
Can you add a comment to `HirId` for posterity, so nobody tries adding these traits in the future?

Reply:
Great idea, @BGR360! Thanks.

Reply:
You can also add a negative impl (`impl !PartialOrd for HirId`) which will cause an error if someone tries to add back a regular impl.

Reply:
Amazing! Thanks @Aaron1011 😇